tests/services/apiService.test.js

```diff
@@ -40,3 +40,87 @@ describe("fetchQuestCategoryData function", () => {
     expect(result).toEqual(mockResponse);
   });
 });
+
+describe("getTasksByQuestId function", () => {
+  beforeEach(() => {
+    fetch.mockClear()
+  });
+
+  it("should handle when api returns no response", async () => {
+    const mockResponse = null;
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    })
+
+    const result = await getTasksByQuestId({ questId: '1', address: '2'});
+    expect(result).toEqual(mockResponse)
+  });
+
+  it("should handle when api return response in unexpected format", async () => {
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve("Unexpected Format")
+    })
+
+    const result = await getTasksByQuestId({ questId: '1', address: '2' })
+    expect(result).not.toEqual(expect.any(Array))
+  });
+
+  it("should handle undefined and null values", async () => {
+    const mockResponse = undefined
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    })
+
+    let result = await getTasksByQuestId({ address: '2' })
```

Comment:
the result here are error strings which the api returns. Not undefined. I would suggest you to test these values in the api and then check if the mock returns you have added here are correct